tst/EnergyPlus/unit/EMSManager.unit.cc

```diff
@@ -2245,3 +2250,339 @@ TEST_F(EnergyPlusFixture, EMS_ViewFactorToGround)
+    }
+
+    auto &argTrendValueErlVar = state->dataRuntimeLang->ErlVariable(3 + numBuiltInErlVars);
+    EXPECT_EQ("ARGTRENDVALUE", argTrendValueErlVar.Name);
+    EXPECT_FALSE(argTrendValueErlVar.ReadOnly);
+    EXPECT_FALSE(argTrendValueErlVar.SetByExternalInterface);
+    EXPECT_ENUM_EQ(DataRuntimeLanguage::Value::Number, argTrendValueErlVar.Value.Type);
+    EXPECT_EQ(0.0, argTrendValueErlVar.Value.Number);
+    EXPECT_TRUE(argTrendValueErlVar.Value.String.empty());
+    EXPECT_EQ(0, argTrendValueErlVar.Value.Expression);
+    EXPECT_TRUE(argTrendValueErlVar.Value.TrendVariable);
+    EXPECT_EQ(1, argTrendValueErlVar.Value.TrendVarPointer);
+    EXPECT_TRUE(argTrendValueErlVar.Value.Error.empty());
+    EXPECT_TRUE(argTrendValueErlVar.Value.initialized);
+
+    auto &trendVar = state->dataRuntimeLang->TrendVariable(1);
+    std::array<Real64, 12> trendValues = {45, 30, 15, 60, 45, 30, 15, 60, 45, 30, 15, 60};
+
+    // rotate right is the same as shift right + insert 60 at first pos in our case
+    // [60, 45, 30, 15, 60, 45, 30, 15, 60, 45, 30, 15]
+    auto trenValuesAfterCallAt60Minute = trendValues;
+    std::rotate(trenValuesAfterCallAt60Minute.rbegin(), trenValuesAfterCallAt60Minute.rbegin() + 1, trenValuesAfterCallAt60Minute.rend());
+    trendVar.TrendValARR = trendValues;
+    EXPECT_EQ("TREND_ARGTRENDVALUE", trendVar.Name);
+    EXPECT_EQ(12, trendVar.LogDepth);
+    EXPECT_EQ(3 + numBuiltInErlVars, trendVar.ErlVariablePointer);
+
+    auto &resultValueErlVar = state->dataRuntimeLang->ErlVariable(4 + numBuiltInErlVars);
+    EXPECT_EQ("RESULTVALUE1", resultValueErlVar.Name);
+    EXPECT_FALSE(resultValueErlVar.ReadOnly);
+    EXPECT_FALSE(resultValueErlVar.SetByExternalInterface);
+    EXPECT_ENUM_EQ(DataRuntimeLanguage::Value::Number, resultValueErlVar.Value.Type);
+    EXPECT_EQ(0.0, resultValueErlVar.Value.Number);
+    EXPECT_TRUE(resultValueErlVar.Value.String.empty());
+    EXPECT_EQ(0, resultValueErlVar.Value.Expression);
+    EXPECT_FALSE(resultValueErlVar.Value.TrendVariable);
+    EXPECT_EQ(0, resultValueErlVar.Value.TrendVarPointer);
+    EXPECT_TRUE(resultValueErlVar.Value.Error.empty());
+    EXPECT_FALSE(resultValueErlVar.Value.initialized); // Note
+
+    bool anyEMSRan = false;
+    EMSManager::ManageEMS(*state, EMSManager::EMSCallFrom::BeginTimestepBeforePredictor, anyEMSRan, ObjexxFCL::Optional_int_const());
+    EXPECT_TRUE(anyEMSRan);
+
+    EXPECT_EQ("ACTUATOR_SCH_DIRECT", actuatorDirectErlVar.Name);
+    EXPECT_FALSE(actuatorDirectErlVar.ReadOnly);
+    EXPECT_FALSE(actuatorDirectErlVar.SetByExternalInterface);
+    EXPECT_ENUM_EQ(DataRuntimeLanguage::Value::Number, actuatorDirectErlVar.Value.Type); // Note: this is Null for an actuator to begin with
+    EXPECT_EQ(45.0, actuatorDirectErlVar.Value.Number);
+    EXPECT_TRUE(actuatorDirectErlVar.Value.String.empty());
+    EXPECT_EQ(0, actuatorDirectErlVar.Value.Expression);
+    EXPECT_FALSE(actuatorDirectErlVar.Value.TrendVariable);
```

Comment:
Before fix, this is actually TRUE, because we assigned a Value  to the actuator that is the result of evaluating a trend variable.

Reply:
Makes sense.  Good fix and test.